alibi_detect/od/_knn.py

```diff
@@ -0,0 +1,221 @@
+from typing import Callable, Union, Optional, Dict, Any, List, Tuple
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from typing_extensions import Literal
+from alibi_detect.base import outlier_prediction_dict
+from alibi_detect.exceptions import _catch_error as catch_error
+from alibi_detect.od.base import TransformProtocol, TransformProtocolType
+from alibi_detect.base import BaseDetector, FitMixin, ThresholdMixin
+from alibi_detect.od.pytorch import KNNTorch, Ensembler
+from alibi_detect.od.base import get_aggregator, get_normalizer, NormalizerLiterals, AggregatorLiterals
+from alibi_detect.utils.frameworks import BackendValidator
+from alibi_detect.version import __version__
+
+
+if TYPE_CHECKING:
+    import torch
+
+
+backends = {
+    'pytorch': (KNNTorch, Ensembler)
+}
+
+
+class KNN(BaseDetector, FitMixin, ThresholdMixin):
+    def __init__(
+        self,
+        k: Union[int, np.ndarray, List[int], Tuple[int]],
+        kernel: Optional[Callable] = None,
+        normalizer: Optional[Union[TransformProtocolType, NormalizerLiterals]] = 'ShiftAndScaleNormalizer',
```

Comment:
I would advocate for the p-value normalizer as default unless you have chosen the shift and scale normalizer for any particular reason?